neon_messagebus_mq_connector/controller.py

```diff
@@ -134,89 +149,144 @@ def handle_neon_profile_update(self, message: Message):
-            if "message_id" in dict_data:
-                dict_data["context"]["mq"]["message_id"] = \
-                    dict_data.pop("message_id")
-            check_error, dict_data = self.validate_request(dict_data)
-            if check_error is not None:
-                LOG.error(check_error)
+            dict_data["context"].setdefault("mq", dict(routing_key=dict_data.pop('routing_key', ''),
+                                                       message_id=dict_data.pop('message_id', '')))
+
+            validation_error, dict_data = self.validate_request(dict_data)
+            if validation_error:
                 response = Message(msg_type="klat.error",
-                                   data=dict(error=str(check_error),
+                                   data=dict(error=validation_error,
                                              message=dict_data))
-                self.handle_neon_message(response, "neon_chat_api_error")
+                response.context.setdefault('klat', {})['routing_key'] = 'neon_chat_api_error'
+                self.handle_neon_message(response)
             else:
+                dict_data["context"].setdefault('ident', f"{dict_data['msg_type']}.response")
                 message = Message(**dict_data)
-                if message.msg_type in ("neon.get_stt", "neon.get_tts",
-                                        "neon.audio_input"):
+                self.bus.emit(message)
+                if message.msg_type in ("neon.get_stt", "neon.get_tts",):
                     # Transactional message, get response
-                    reply_type = message.context.get("ident")
-                    response = self.bus.wait_for_response(message, reply_type,
-                                                          timeout=30)
-                    # Replace response message type for MQ client to handle
-                    response.msg_type = f'{message.msg_type}.response'
-                    response = response or \
-                        message.response(data={"success": False,
-                                               "error": "no response"})
-                    self.handle_neon_message(response)
-                else:
-                    # Probable user input to generate klat.response message
-                    self.bus.emit(message)
-
-            channel.basic_ack(method.delivery_tag)
+                    self.push_awaiting_message(message)
+            channel.basic_ack()
         else:
             channel.basic_nack()
             raise TypeError(f'Invalid body received, expected: bytes string;'
                             f' got: {type(body)}')
+
+    def format_response(self, response_type: str, message: Message) -> dict:
+        """ Formats received STT response by Neon API based on type """
+        message_id = message.context.get('mq', {}).get('message_id')
+        matching_message_data = self.awaiting_messages.get('stt', {}).get(message_id)
```

Comment:
Below (Line 271) looks like there's TTS handling in this method too; maybe `awaiting_messages` should just be a flat list of all requests?

Reply:
Also, consider just adding data to message context instead of the internal `awaiting_messages` dict (i.e. `message.context['klat_data']['request'] = {'lang': '', 'gender': '', 'created_on': ''}`)